source/MapDetailPanel.cpp

```diff
@@ -853,11 +874,44 @@ void MapDetailPanel::DrawInfo()
 		else
 			price = (canView ? "n/a" : "?");
 
-		const auto alignRight = Layout(140, Alignment::RIGHT, Truncate::BACK);
+		const auto alignRight = Layout(130, Alignment::RIGHT, Truncate::BACK);
 		font.Draw({price, alignRight}, uiPoint, color);
 
 		if(isSelected)
-			PointerShader::Draw(uiPoint + Point(0., 7.), Point(1., 0.), 10.f, 10.f, 0.f, color);
+			// const Point &center, const Point &angle, float width, float height, float offset, const Color &color
+				PointerShader::Draw(uiPoint + Point(0., 7.), Point(1., 0.), 10.f, 10.f, 0.f, color);
+
+		// Draw colored icons, when values are displayed.
+		if(canView && selectedSystem->IsInhabited(player.Flagship()))
+		{
+			if(!noCompare && player.GetSystem() != selectedSystem)
+			{
+				// Determine the relative negative-ness or positive-ness of the value compared to low/high.
+				// Note: if value is negative, lowCompare will be negative and if value is positive, highCompare will be
+				//       positive.
+				double v = 0;
+				if(value < 0)
+					v = static_cast<double>(value) / abs(lowCompare);
+				else if(value > 0)
+					v = static_cast<double>(value) / highCompare;
+				color = MapColor(v);
+				// Draw up/down/equals arrows based on price delta (value).
+				PointerShader::Draw(uiPoint + Point(143, 7. + (-7 * v)), Point(0., 1), 20.f,
+					static_cast<float>(-14. * v), 0.f, color);
```

Comment:
Can't we just do this? And then we could leave the color as a reference.
```suggestion
				// Draw up/down/equals arrows based on price delta (value).
				PointerShader::Draw(uiPoint + Point(143, 7. + (-7 * v)), Point(0., 1), 20.f,
					static_cast<float>(-14. * v), 0.f, MapColor(v));
```